shell/platform/darwin/common/framework/Source/FlutterNSBundleUtils.mm

```diff
@@ -48,23 +49,41 @@
     flutterFrameworkBundle = [NSBundle bundleWithIdentifier:flutterFrameworkBundleID];
   }
   if (flutterFrameworkBundle == nil) {
-    flutterFrameworkBundle = [NSBundle mainBundle];
+    flutterFrameworkBundle = NSBundle.mainBundle;
   }
   return flutterFrameworkBundle;
 }
 
 NSString* FLTAssetPath(NSBundle* bundle) {
-  return [bundle objectForInfoDictionaryKey:@"FLTAssetsPath"] ?: kDefaultAssetPath;
+  NSString* relativeAssetsPath = [bundle objectForInfoDictionaryKey:@"FLTAssetsPath"];
+  if (relativeAssetsPath.length == 0) {
+    // Try main bundle
+    relativeAssetsPath = [NSBundle.mainBundle objectForInfoDictionaryKey:@"FLTAssetsPath"];
+  }
+  if (relativeAssetsPath.length == 0) {
+    relativeAssetsPath = kDefaultAssetPath;
+  }
+  return relativeAssetsPath;
 }
 
 NSString* FLTAssetsPathFromBundle(NSBundle* bundle) {
-  NSString* flutterAssetsPath = FLTAssetPath(bundle);
```

Comment:
The issue was this method was called twice even if `[bundle objectForInfoDictionaryKey:@"FLTAssetsPath"]` is not nil.